static/panes/editor.ts

```diff
@@ -77,7 +78,7 @@ export class Editor extends MonacoPane<monaco.editor.IStandaloneCodeEditor, Edit
     private prevDecorations: string[];
     private extraDecorations?: Decoration[];
     private fadeTimeoutId: NodeJS.Timeout | null;
-    private editorSourceByLang: Record<LanguageKey, string | undefined>;
+    private editorSourceByLang: Partial<Record<LanguageKey, string | undefined>>;
```

Comment:
So interestingly `Record<"a" | "b", string>` is the same as `{a: string, b: string}` but really here we want a dictionary with a limited set of keys

Reply:
hence the partial? as some of the languages are optional?

Reply:
I believe so yeah, the dictionary is populated over time